lib/logstash/inputs/lumberjack.rb

```diff
@@ -31,10 +31,13 @@ class LogStash::Inputs::Lumberjack < LogStash::Inputs::Base
 
   # This setting no longer has any effect and will be removed in a future release.
   config :max_clients, :validate => :number, :deprecated => "This setting no longer has any effect. See https://github.com/logstash-plugins/logstash-input-lumberjack/pull/12 for the history of this change"
+  
+  # The number of seconds before we raise a timeout,
+  # this option is useful to control how much time to wait if something is blocking the pipeline.
+  config :timeout, :validate => :number, :default => 5
 
   # TODO(sissel): Add CA to authenticate clients with.
-
-  BUFFERED_QUEUE_SIZE = 20
+  BUFFERED_QUEUE_SIZE = 1
```

Comment:
By using 1, we reduce the number of duplicates we could possibly have.
Also by using 1, it's closer to the real status of the pipeline. 

Not sure we will get great gain if we have higher values because of the lock usage.